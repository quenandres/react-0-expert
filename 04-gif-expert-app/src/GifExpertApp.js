import React, { useState } from 'react'
import { AddCategory } from './components/AddCategory'

export const GifExpertApp = () => {
    
    //const categories = ['One punch', 'Samurai x', 'Dragon ball'];
    
    const [categories, setCategories] = useState(['One punch', 'Samurai x', 'Dragon ball'])

    /* const handleAdd = () => {
        const anime = 'Hunter x hunter';
        setCategories( cats => [...cats, anime] );
    } */

    return (
        <div>
            <>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories={ setCategories }/>
            <hr />
            {/* <button onClick={handleAdd}>Agregar</button> */}

            <ol>
                {
                    categories.map( (category) => {
                        return <li key={ category }>{ category }</li>
                    })
                }
            </ol>
            </>
        </div>
    )
}
