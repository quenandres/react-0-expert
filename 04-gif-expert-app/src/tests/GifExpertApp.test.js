import React from 'react';
import { shallow } from 'enzyme';
import { GifExpertApp } from '../GifExpertApp';
import '@testing-library/jest-dom';

describe('Pruebas en GifExpertApp', () => {
    test('debe mostrarse correctamente el GifExpertApp', () => {
        const wrapper = shallow(<GifExpertApp />);
        expect( wrapper ).toMatchSnapshot();
    });

    test('debe de mostrar una lista de categorias', () => {
        const categories = ['One punch', 'Drangon ball'];
        const wrapper = shallow(<GifExpertApp defaultCategories={ categories } />);
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( categories.length );
    });
})
