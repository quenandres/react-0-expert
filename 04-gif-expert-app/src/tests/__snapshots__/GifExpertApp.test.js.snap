// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifExpertApp debe de mostrar una lista de categorias 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One punch"
      key="One punch"
    />
    <GifGrid
      category="Drangon ball"
      key="Drangon ball"
    />
  </ol>
</div>
`;

exports[`Pruebas en GifExpertApp debe mostrarse correctamente el GifExpertApp 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</div>
`;
